src/api/app/models/user.rb

```diff
@@ -871,6 +871,18 @@ def run_as
     end
   end
 
+  def watchlist_projects
```

Comment:
This method and the others below could be in a query object. This way, we wouldn't increase the size of the `User` model which is already quite large with almost 900+ lines of code.